rust/frontend/src/server.rs

```diff
@@ -445,6 +568,21 @@ async fn list_databases(
     Ok(Json(server.frontend.list_databases(request).await?))
 }
 
+/// Retrieves a specific database by name.
+#[utoipa::path(
+    get,
+    path = "/api/v2/tenants/{tenant_id}/databases/{database_name}",
+    responses(
+        (status = 200, description = "Database retrieved successfully", body = GetDatabaseResponse),
+        (status = 401, description = "Unauthorized", body = ErrorResponse),
+        (status = 404, description = "Database not found", body = ErrorResponse),
```

Comment:
I wish it were possible to infer this from the response type :(

Reply:
I dug in, and it seems that the way we're doing typed returns elsewhere are making it hard right now. I propose we add it as a follow-up task.